usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
zfs on linux could have issues since it currently has a limited queue size for events.  Perhaps we could add a way to opt in/out of `ESC_ZFS_POOL_HISTORY` event posting?